File: csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    public static string Response(string statement)
    {
        if (string.IsNullOrWhiteSpace(statement))
            return "Fine. Be that way!";
        if (statement.Any(Char.IsLetter) && statement.Where(Char.IsLetter).All(Char.IsUpper))
            if (statement.Contains("?"))
                return "Calm down, I know what I'm doing!";
            else
                return "Whoa, chill out!";
        if (statement.Trim().EndsWith("?"))
            return "Sure.";

        return "Whatever.";
    }
}

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.trim(input) in [nil, ""] ->
        "Fine. Be that way!"

      Regex.match?(~r/[a-zA-Z]/, input) && input == String.upcase(input) ->
        if String.contains?(input, "?"),
          do: "Calm down, I know what I'm doing!",
          else: "Whoa, chill out!"

      input |> String.trim() |> String.ends_with?("?") ->
        "Sure."

      true ->
        "Whatever."
    end
  end
end
